common/src/types/v0/store/node.rs

```diff
@@ -60,11 +68,29 @@ impl NodeSpec {
     pub fn set_endpoint(&mut self, endpoint: String) {
         self.endpoint = endpoint
     }
+    /// Cordon node by applying the label.
+    pub fn cordon(&mut self, label: String) {
+        self.cordon_labels.push(label);
+    }
+    /// Uncordon node by removing the corresponding label.
+    pub fn uncordon(&mut self, label: String) {
+        if let Ok(index) = self.cordon_labels.binary_search(&label) {
+            self.cordon_labels.remove(index);
+        }
```

Comment:
Maybe this should return an error if the label cannot be found (rather than searching before calling this function)